virtualmachine/vsphere/vm.go

```diff
@@ -1899,3 +1899,47 @@ func (vm *VM) ValidateAuth() error {
 	defer vm.cancel()
 	return nil
 }
+
+// GetFilteredHosts: Lists the hosts compatible with the network and datastores
+func GetFilteredHosts(vm *VM) ([]string, error) {
```

Comment:
It is understood that we are using filterHosts method at core and thus named it GetFilteredHosts, but will renaming it to compatibleHosts & GetCompatibleHosts make more sense?